### PR TITLE
Protein translation (fix #267)

### LINKED ISSUE #267: Tests for protein-translation don't match its README
The [README](http://exercism.io/exercises/rust/protein-translation/readme) for this exercise describe the problem as (roughly) "given a sequence of codons, output the sequence of proteins". However, the tests in `protein-translation` (a) only seem to expect a single protein name from a single codon and (b) talk about codon "compression" which is not mentioned in the README; e.g.,

```rust
fn test_cysteine_tgy() { // "compressed" name for TGT and TGC
    let info = proteins::parse(make_pairs());
    assert_eq!(info.name_for("TGT"), info.name_for("TGY"));
    assert_eq!(info.name_for("TGC"), info.name_for("TGY"));
}

...

fn test_arginine_name() {
    // In arginine CGA can be "compresed" both as CGN and as MGR
    let info = proteins::parse(make_pairs());
    assert_eq!(info.name_for("CGA"), Ok("arginine"));
    assert_eq!(info.name_for("CGN"), Ok("arginine"));
    assert_eq!(info.name_for("MGR"), Ok("arginine"));
}
```

I glanced at a couple other languages' versions of this exercise and saw tests that look much more like the README and don't talk about compressed names ([go](https://github.com/exercism/xgo/blob/master/exercises/protein-translation/protein_translation_test.go), [elixir](https://github.com/exercism/xelixir/blob/master/exercises/protein-translation/protein_translation_test.exs)).

### PR DESCRIPTION
**Pull request by go717franciswang**
Split the earlier pull request into 2 commits -- remove tests for codon compression, and add tests for `of_rna`. It resolves #267.

### CODE REVIEW

**Comment by petertseng** [comment]
Nice, looks like it's more understandable as two commits.

An interesting thing happened. The https://github.com/exercism/x-common/blob/master/exercises/protein-translation/description.md is talking about RNA (ACGU), but nucleotide-codons (https://github.com/exercism/x-common/blob/master/exercises/nucleotide-codons/description.md) used to talk about DNA (ACGT) before #258 blindly renamed nucleotide-codons to protein-translation.

It would probably be for the best if all the T were changed to U, that's my fault. You can do it if you like otherwise I will have to soon.

**Comment by go717franciswang** [comment]
Ah. I see. I referenced [wikipedia](https://en.wikipedia.org/wiki/Genetic_code#RNA_codon_table), and switched the DNA codons with RNA codons in a new commit.

File: exercises/protein-translation/tests/proteins.rs [hunk 1]
```diff
@@ -3,42 +3,41 @@ extern crate protein_translation as proteins;
...
     let info = proteins::parse(make_pairs());
-    assert_eq!(info.name_for("ATT"), Ok("isoleucine"));
+    assert_eq!(info.name_for("AUU"), Ok("isoleucine"));
 }
 
 #[test]
 #[ignore]
 fn test_arginine_name() {
     let info = proteins::parse(make_pairs());
-    assert_eq!(info.name_for("CGA"), Ok("arginine"));
```
Comment (petertseng) [inline exercises/protein-translation/tests/proteins.rs]: removal intentional? I notice `CGA` is still an arginine in https://en.wikipedia.org/wiki/Genetic_code#RNA_codon_table

Comment (go717franciswang) [inline exercises/protein-translation/tests/proteins.rs]: My bad. That was a mistake. Let me add the missing ones.

**Comment by go717franciswang** [comment]
Done. The last 2 commits were squashed.

**Comment by petertseng** [comment]
All right. Thanks for the work!

**Comment by go717franciswang** [comment]
Awesome. Thanks!

**Pull request merged**